main.py

```diff
@@ -52,24 +52,199 @@
 logger.debug("Finished get_all_currencies")
 
 
+def buy():
+    while not globals.stop_threads:
+        logger.debug('Waiting for buy_ready event')
+        globals.buy_ready.wait()
+        logger.debug('buy_ready event triggered')
+        globals.buy_ready.clear()
+        if globals.stop_threads:
+            break
+        # the buy block and logic pass
```

Comment:
This line and on are straight copy/paste from main (except for and `globals.` before the now-global variables)